.github/workflows/integration_tests.yaml

```diff
@@ -0,0 +1,97 @@
+          JENKINS_PASSWORD: ${{ secrets.JENKINS_API_TOKEN }}
+          REPO: ${{ github.event.repository.name }}
+          BRANCH: ${{ github.event.pull_request.head.ref }}
+          PR_NUM: ${{ github.event.pull_request.number }}
+          PR_LABELS: ${{ toJson(github.event.pull_request.labels.*.name) }}
+          COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+          COMMIT_TIME: ${{ github.event.pull_request.head.repo.pushed_at }}
+
+        steps:
+          # Checkout current branch of gen3-code-vigil if it is the repo under test
+          - name: Checkout integration test code from dev branch
+            if: github.event.repository.name == 'gen3-code-vigil'
+            uses: actions/checkout@v4
+
+          # Checkout master branch of gen3-code-vigil when another repo is under test
+          - name: Checkout integration test code from master branch
+            if: github.event.repository.name  != 'gen3-code-vigil'
+            uses: actions/checkout@v4
+            with:
+              repository: uc-cdis/gen3-code-vigil
+              ref: master
+
+          # gen3-integration-tests run with python 3.9
+          - name: Set up Python
+            uses: actions/setup-python@v4
+            with:
+              python-version: 3.9
+
+          # Install gen3-integration-tests dependencies
+          - name: Install dependencies
+            run: |
+              python -m pip install --upgrade pip
+              pip install poetry
+              poetry install -vv
+
+          # TODO: Rely on a database in AWS to make this faster
+          # Select an unlocked environment
+          # If an env is specified in a PR label use it, else pick one from the pool
+          - name: Select CI environment
+            run: |
+              labels="${PR_LABELS//\\n/ }"
+              env_label=$(echo "$labels" | jq -r 'map(select(test("jenkins-"; "i"))) | first')
+              if [[ $env_label != "" && $env_label != null ]]; then
+                echo "Found PR label $env_label"
+                poetry run python -m gen3_ci.scripts.select_ci_environment $env_label
+              else
+                poetry run python -m gen3_ci.scripts.select_ci_environment
+              fi
+
+          # TODO: Improve the logic to do differential updates to the env, not roll all services
+          # Apply the changes to the manifest of the selected CI environment, roll the pods and run usersync
+          # Generate API keys for test users for the environment
+          - name: Prepare CI environment
+            run: |
+              mkdir $HOME/.gen3
+              poetry run python -m gen3_ci.scripts.prepare_ci_environment
+
+          - name: Run tests
+            run: poetry run pytest --html=output/report.html --self-contained-html -n auto -m "not wip"
+
+          - name: Archive test report
+            uses: actions/upload-artifact@v3
+            with:
+              name: test-report
+              path: ~/**/output/report.html
```

Comment:
at the moment this isn't a super useful report (just tests passed/failed)

One thing to consider is how we mimic some of features of current Jenkins integration testing (like screenshots for UI tests and getting a dump of the service logs for when things go wrong).

I don't think you have to solve that right now, but we will want a way to easily get to service logs when the integration tests fail

Reply:
Not sure if you noticed the `show details` next to the result ... there are step logs as well in there once expanded. But yes, still looking for a better reporting mechanism.